cmd/benchmark.go

```diff
@@ -0,0 +1,329 @@
+// Copyright © Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Azure/azure-storage-azcopy/common"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/spf13/cobra"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+// represents the raw benchmark command input from the user
+type rawBenchmarkCmdArgs struct {
+	// no src, since it's implicitly the auto-data-generator used for benchmarking
+
+	// where are we uploading the benchmark data to?
+	dst string
+
+	// parameters controlling the auto-generated data
+	sizePerFile    string
+	fileCount      uint
+	deleteTestData bool
+
+	// options from flags
+	blockSizeMB  float64
+	putMd5       bool
+	blobType     string
+	output       string
+	logVerbosity string
+}
+
+const (
+	maxBytesPerFile = 4.75 * 1024 * 1024 * 1024 * 1024
+
+	// TODO would it be better to have a trailing B, eg. 12KB or 200GB? (that might make it case sensitive, or at least
+	//    necessitate making the B case sensitive, because lowercase b means bits (and we don't want to bother supporting bits)
+	sizeStringDescription = "a number immediately followed by K, M or G. E.g. 12k or 200G"
+)
+
+// TODO move to copy handler util
```

Comment:
we should get rid of the util file or at least not add to it. It became a dumping ground for unrelated code.